src/Form/PasswordConstraintFormBase.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace Drupal\password_enhancements\Form;
+
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Form\SubformState;
+use Drupal\Component\Plugin\Exception\PluginNotFoundException;
+use Drupal\password_enhancements\PasswordConstraintInterface;
+use Drupal\password_enhancements\PasswordPolicy;
+use Drupal\user\RoleInterface;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * Provides a base form for password constraints.
+ */
+abstract class PasswordConstraintFormBase extends FormBase {
+
+  /**
+   * The password policy.
+   *
+   * @var \Drupal\password_enhancements\PasswordPolicyInterface
+   */
+  protected $policy;
+
+  /**
+   * The password constraint.
+   *
+   * @var \Drupal\password_enhancements\PasswordConstraintInterface
+   */
+  protected $constraint;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
```

Comment:
This should not be defined in the parent class because all child forms should have a unique form id.  (The base form id of these forms although could be the same, not sure whether it is automatically set by the form API or how you can specify that in this class.)

```
  /**
   * Returns a unique string identifying the form.
   *
   * The returned ID should be a unique string that can be a valid PHP function
   * name, since it's used in hook implementation names such as
   * hook_form_FORM_ID_alter().
   *
   * @return string
   *   The unique string identifying the form.
   */
  public function getFormId();
```